cluster_pack/skein/skein_helper.py

```diff
@@ -30,15 +30,35 @@ def get_application_logs(
     return None
 
 
-def wait_for_finished(client: skein.Client, app_id: str):
-    logger.info(f"application_id: {app_id}")
+def wait_for_finished(client: skein.Client, app_id: str, poll_every_secs: int = 5):
+    logger.info(f"waiting for application_id: {app_id}")
+    state = None
     while True:
         report = client.application_report(app_id)
 
-        logger.info(report)
+        logger.info(
+            f"Application report for {app_id} (state: {report.state})")
+        if state != report.state:
+            logger.info(_format_app_report(report))
 
         if report.final_status != "undefined":
             logger.info(report.final_status)
-            break
+            return report.final_status == "succeeded"
```

Comment:
It is the a ApplicationState returned here ? Maybe we can use this constant to do the comparison: https://jcristharif.com/skein/api.html#application-responses

Reply:
I was afraid of the linter as it is defined liek this https://github.com/jcrist/skein/blob/master/skein/model.py#L505 but it seems fine.